src/librustc_mir/interpret/place.rs

```diff
@@ -941,8 +941,14 @@ where
             return self.copy_op(src, dest);
         }
         // We still require the sizes to match.
-        assert!(src.layout.size == dest.layout.size,
-            "Size mismatch when transmuting!\nsrc: {:#?}\ndest: {:#?}", src, dest);
+        if src.layout.size != dest.layout.size {
+            // FIXME: This should be an assert instead of an error, but if we transmute within an
+            // array length computation, `typeck` may not have yet been run and errored out. In fact
+            // most likey we *are* running `typeck` right now. Investigate whether we can bail out
+            // on `typeck_tables().has_errors` at all const eval entry points.
+            debug!("Size mismatch when transmuting!\nsrc: {:#?}\ndest: {:#?}", src, dest);
+            throw_unsup!(TransmuteSizeDiff(src.layout.ty, dest.layout.ty));
```

Comment:
`delay_span_bug`? Or can typeck still succeed when hitting this?

Reply:
The problem is that we can't possibly continue evaluation here, there's no useful operation to do. So we need to error out. I'd rather fix this correctly in a follow up where I enforce that we don't even try to evaluate if we have typeck errors